gluten-core/src/main/scala/org/apache/spark/util/memory/TaskMemoryResources.scala

```diff
@@ -84,12 +87,12 @@ object TaskMemoryResources extends Logging {
     }
   }
 
-  def addLeakSafeTaskCompletionListener[U](f: TaskContext => U): TaskContext = {
+  def addRecycler(priority: Long)(f: TaskContext => Unit): Unit = {
```

Comment:
Can we have a new function `addRecycler()` with default value 100?

Reply:
Yes we can but personally I don't suggest doing that because once one calls this API, he/she should better know exactly what priority means. If we hide this detail from users they may not be able to manage the resources appropriately.

Reply:
For most case, the listener has same priority, it is just some special case that it has different priority. Only the user who has special need should ko=now the meaning of priority. What do you think? @FelixYBW

Reply:
Since patch is somehow important for spilling, let's move the discussion into https://github.com/oap-project/gluten/pull/1527